maintenance/projects/js-toolkit/packages/liferay-npm-build-tools-common/src/project/index.ts

```diff
@@ -384,36 +397,56 @@ export class Project {
 
 			return autoPresets;
 		}, []);
-	}
 
-	_loadDefaultPreset(): string {
-		let presetFilePath;
-
-		const autoPresets = this._findAutopresets();
-
-		if (autoPresets.length > 1) {
+		if (autopresets.length > 1) {
 			throw new Error(
 				'Multiple autopreset dependencies found in project (' +
-					autoPresets +
+					autopresets +
 					'): please remove the invalid ones or ' +
 					'explicitly define the preset to be used in the ' +
 					'.npmbundlerrc file'
 			);
 		}
-		else if (autoPresets.length) {
-			presetFilePath = this.resolve(autoPresets[0]);
+
+		return autopresets.length ? autopresets[0] : undefined;
```

Comment:
I'm leaning more towards `undefined` for these things because, well, the preset it's truly `not defined` :sweat_smile:, but I agree that JavaScript, being one of the only languages with that distinction, makes it a bit harder to know when to use which.

So, my rule of thumb is to use `null` when something is explicitly disabled/missing, and `undefined` when it is simply missing (we know it's not there, but nobody told us explicitly).

So, `null` is a strong hint that a configuration value is disabled, while `undefined` is missing and it can be replaced by a default value or a fallback, for example.

Look, [this stackoverflow answer](https://stackoverflow.com/questions/12606306/es5-when-to-use-null-and-when-to-use-undefined#12609664) makes it even more difficult :joy: . It says:

> The common sense is the following: undefined are variables which never have been assigned and non-existing properties.
> null however is a state of a variable or property which indicates that it has no value assigned.

So, what I'm saying, more or less. But then, there's a plot twist for function returns:

> Some methods like getElement... explicitely return null to indicate that the resultset is empty. If your function has no return statement, implicitely undefined is returned instead.
> In general, always assign null and never undefined.

Which I admit makes sense, since "void" functions return `undefined`.

So, I would say that my reasoning makes sense for variable assignments and properties, but not for function returns, as you say. 

Does that make sense? :thinking:

Reply:
Addressed in https://github.com/liferay/liferay-frontend-projects/pull/576